FRC-Extension/Buttons/DeployDebugButton.cs

```diff
@@ -55,27 +57,30 @@ private void EnableAllButtons()
                 try
                 {
                     Output.ProgressBarLabel = "Deploying Robot Code";
-                    OutputWriter.Instance.Clear();
-                    string teamNumber = Package.GetTeamNumber();
+                    await OutputWriter.Instance.ClearAsync().ConfigureAwait(false);
+                    string teamNumber = await Package.GetTeamNumberAsync().ConfigureAwait(false);
 
                     if (teamNumber == null) return;
 
                     //Disable the deploy buttons
+                    await ThreadHelperExtensions.SwitchToUiThread();
                     DisableAllButtons();
-                    DeployManager m = new DeployManager(Package.PublicGetService(typeof (DTE)) as DTE);
-                    bool success = await m.DeployCode(teamNumber, DebugButton, m_robotProject);
+                    DeployManager m = new DeployManager(Package.PublicGetService<DTE>());
+                    bool success = await m.DeployCodeAsync(teamNumber, DebugButton, m_robotProject).ConfigureAwait(true);
                     EnableAllButtons();
                     Output.ProgressBarLabel = success ? "Robot Code Deploy Successful" : "Robot Code Deploy Failed";
                 }
                 catch (SshConnectionException)
                 {
-                    Output.WriteLine("Connection to RoboRIO lost. Deploy aborted.");
+                    await Output.WriteLineAsync("Connection to RoboRIO lost. Deploy aborted.").ConfigureAwait(false);
+                    await ThreadHelperExtensions.SwitchToUiThread();
                     EnableAllButtons();
```

Comment:
For things like this, shouldn't we put the SwitchToUiThread in the EnableAllButtons function, and make that awaitable? Seems cleaner and less duplicated code.